### PR TITLE
Add course information to CourseList

### LINKED ISSUE #134: Course Search — Pre/Coreq and description info
**Job**
When I'm searching for issues, I want to be able to see the course description and pre/co-reqs

**Describe a possible solution**
Show a (info) button that drops down extended course info

**Context**
This data is in the api all response in /class

### PR DESCRIPTION
**Pull request by 123joshuawu**
**Issue**

Closes #134 

**Database Changes/Migrations**

N/A

**Test Modifications**
N/A

**Test Procedure**

Click on information icon to see course information including if available: coreqs, prereqs, and description

**Photos**

Screen Shot 2020-04-24 at 4 35 23 PM
Screen Shot 2020-04-25 at 12 34 37 PM
Screen Shot 2020-04-25 at 12 34 11 PM
Screen Shot 2020-04-25 at 12 33 54 PM
Screen Shot 2020-04-25 at 12 33 23 PM
Screen Shot 2020-04-25 at 12 33 12 PM

**Additional Info**
Refactor CourseListing to use slots. CourseList now overrides default slots to show course information

### CODE REVIEW

**Comment by 123joshuawu** [comment]
will probably want to get #131 merged in before this one

File: src/web/src/components/CourseList.vue [hunk 1]
```diff
@@ -28,12 +28,33 @@
...
+          <template #toggleCollapseButton="{ course, toggleCollapse }">
+            <button
+              class="btn"
+              @click="toggleCollapse()"
+              :disabled="!course.corequisites && !course.prerequisites"
+            >
+              <font-awesome-icon :icon="faInfoCircle" />
+            </button>
+          </template>
+          <template #collapseContent="{ course: {corequisites, prerequisites, description} }">
```
Comment (CoreyBuckley) [inline src/web/src/components/CourseList.vue:46]: What does the hash sign do in this case?

File: src/web/src/components/CourseList.vue [hunk 2]
```diff
@@ -28,12 +28,33 @@
...
         :key="course.id"
         :class="{'bg-light': course.selected}"
       >
-        <CourseListing :course="course" :actions="{add:true}" v-on="$listeners" />
+        <CourseListing :course="course" lazyLoadCollapse v-on="$listeners">
+          <template #toggleCollapseButton="{ course, toggleCollapse }">
+            <button
+              class="btn"
+              @click="toggleCollapse()"
+              :disabled="!course.corequisites && !course.prerequisites"
```
Comment (CoreyBuckley) [inline src/web/src/components/CourseList.vue:41]: && !course.raw_precoreqs

File: src/web/src/components/CourseList.vue [hunk 3]
```diff
@@ -28,12 +28,33 @@
...
+            >
+              <font-awesome-icon :icon="faInfoCircle" />
+            </button>
+          </template>
+          <template #collapseContent="{ course: {corequisites, prerequisites, description} }">
+            <span v-if="corequisites">
```
Comment (CoreyBuckley) [inline src/web/src/components/CourseList.vue:47]: i'm not sure what the condition should be, but show raw_precoreqs if the parsed out prereqs is empty?

**Comment by CoreyBuckley** [comment]
Am getting 502 on temp server ☹️ 
 image

File: src/web/src/components/CourseList.vue (hunk 1 above)
Comment (123joshuawu) [reply src/web/src/components/CourseList.vue:46]: I believe its a shorthand for named slots so the alternative is `v-slot:name=`

File: src/web/src/components/CourseList.vue (hunk 3 above)
Comment (123joshuawu) [reply src/web/src/components/CourseList.vue:47]: yeah sounds good to me

File: src/web/src/components/CourseList.vue (hunk 2 above)
Comment (123joshuawu) [reply src/web/src/components/CourseList.vue:41]: claro que si senor

**Comment by CoreyBuckley** [comment]
The clickable space for adding a class seems to be obscuring the info btn.  Whenever I try to click the info btn, it just adds the course instead.

**Comment by 123joshuawu** [comment]
@CoreyBuckley just tried the temp deploy and seems to be working there
not sure why that would happen because i think events usually propagate from the child up

**Comment by CoreyBuckley** [comment]
I was being dumb. Was clicking it on a class where it was disabled.

_Therefore_, I propose you get rid of the info button in place of it being disabled. Just leads to confusion.

**Comment by jshom** [comment]
> I was being dumb. Was clicking it on a class where it was disabled.
> 
> _Therefore_, I propose you get rid of the info button in place of it being disabled. Just leads to confusion.

Agree

**Comment by jshom** [comment]
Screen Shot 2020-04-24 at 9 29 21 PM

^ Looks a bit bland, lets add some spice

- Lets use the full_title instead of title
- Lets maybe use language of Require completion of CSCI-2500, 2300 instead of pre-req: x,y,z.
- Also, lets collect the additional info in a box of some sort, maybe block quote or some other way of grouping the additional info visually

**Comment by 123joshuawu** [comment]
Sounds good, lemme put those in

**Pull request merged**